server/src/main/java/org/opensearch/snapshots/RestoreService.java

```diff
@@ -806,6 +829,40 @@ private IndexMetadata updateIndexSettings(
                         return builder.settings(settingsBuilder).build();
                     }
 
+                    private void validateSearchableSnapshotRestorable(long totalRestorableRemoteIndexesSize) {
+                        ClusterInfo clusterInfo = clusterInfoService.getClusterInfo();
+                        Map<String, FileCacheStats> nodeFileCacheStats = clusterInfo.getNodeFileCacheStats();
+                        if (nodeFileCacheStats.isEmpty()) {
+                            return;
+                        }
+
+                        long totalNodeFileCacheSize = clusterInfo.getNodeFileCacheStats()
+                            .values()
+                            .stream()
+                            .map(fileCacheStats -> fileCacheStats.getTotal().getBytes())
+                            .mapToLong(Long::longValue)
+                            .sum();
+
+                        long totalRestoredRemoteIndexesSize = 0;
+                        for (IndexService indexService : indicesService) {
+                            if (indexService.getIndexSettings().isRemoteSnapshot()) {
+                                for (IndexShard indexShard : indexService) {
+                                    if (indexShard.routingEntry().primary()) {
```

Comment:
Should we move this to RoutingTable to ensure this is more re-usable for fetching all shards with a particular settings like remote snapshot, segrep, remote store etc?